src/app/static/src/js/components/runWorkflow/workflowParametersMixin.ts

```diff
@@ -0,0 +1,77 @@
+import Vue from "vue";
+import {api} from "../../utils/api";
+import {Parameter, WorkflowSummary} from "../../utils/types";
+import {AxiosResponse} from "axios";
+
+interface WorkflowReportParams {
+    defaultParams: Parameter[],
+    nonDefaultParams: Parameter[]
+}
+
+interface Data {
+    workflowReportParams: WorkflowReportParams[] | null;
+    defaultParamsErrors: Record<string, Error>
+}
+
+interface Methods {
+    getParametersApiCall: (reportName: string, gitCommit: string) => Promise<AxiosResponse>;
+    getWorkflowReportParams: (workflowSummary: WorkflowSummary, gitCommit: string) => void
+}
+
+export default Vue.extend<Data, Methods, unknown, unknown>({
+    data() {
+        return {
+            workflowReportParams: null,
+            defaultParamsErrors: {}
+        }
+    },
+    methods: {
+        getParametersApiCall(reportName, gitCommit) {
+            const commit = gitCommit ? `?commit=${gitCommit}` : '';
+            return api.get(`/report/${reportName}/config/parameters/${commit}`);
+        },
+        // For each report in workflowSummary, map to an object which contains both defaultParams and nonDefaultParams
+        getWorkflowReportParams: async function (workflowSummary: WorkflowSummary, gitCommit: string) {
+            const workflowReportParams = [];
```

Comment:
It feels to me like all of this fiddly and hard to read logic (so many nested loops!) should be implemented in the back-end and a more appropriate json object returned for use by the front-end. As well as being easier to write and reason about in kotlin, it will be more performant than having the browser make an extra api call per report to get the default parameters. 

The idea would be to change the returned reports slightly to:

```
WorkflowReportWithDependencies {
    name: string,
    instance?: string
    defaultParams?: Record<string, string>,
    nonDefaultParams?: Record<string, srtring>,
    depends_on?: string[]
}
```

and then the component looks much simpler.